admin/code/CampaignAdmin.php

```diff
@@ -0,0 +1,408 @@
+			"customValidationMessage": "",
+			"attributes": [],
+			"data": []
+		}],
+		"actions": []
+	}
+}
+JSON;
+
+		$formName = $request->param('ID');
+		if($formName == 'EditForm') {
+			$response = $this->getResponse();
+			$response->addHeader('Content-Type', 'application/json');
+			$response->setBody($json);
+			return $response;
+		} else {
+			return parent::schema($request);
+		}
+	}
+
+	/**
+	 * REST endpoint to create a campaign.
+	 *
+	 * @param SS_HTTPRequest $request
+	 *
+	 * @return SS_HTTPResponse
+	 */
+	public function createCampaign(SS_HTTPRequest $request) {
+		$response = new SS_HTTPResponse();
+		$response->addHeader('Content-Type', 'application/json');
+		$response->setBody(Convert::raw2json(['campaign' => 'create']));
+
+		return $response;
+	}
+
+	/**
+	 * REST endpoint to get a list of campaigns.
+	 *
+	 * @param SS_HTTPRequest $request
+	 *
+	 * @return SS_HTTPResponse
+	 */
+	public function readCampaigns(SS_HTTPRequest $request) {
+		$response = new SS_HTTPResponse();
+		$response->addHeader('Content-Type', 'application/json');
+		$hal = $this->getListResource();
+		$response->setBody(Convert::array2json($hal));
+		return $response;
+		}
+
+	/**
+	 * Get list contained as a hal wrapper
+	 *
+	 * @return array
+	 */
+	protected function getListResource() {
+		$items = $this->getListItems();
+		$count = $items->count();
+		$hal = [
+			'count' => $count,
+			'total' => $count,
+			'_links' => [
+				'self' => [
+					'href' => $this->Link('items')
+				]
+			],
+			'_embedded' => ['ChangeSets' => []]
+		];
+		foreach($items as $item) {
+			/** @var ChangeSet $item */
+			$resource = $this->getChangeSetResource($item);
+			$hal['_embedded']['ChangeSets'][] = $resource;
+		}
+		return $hal;
+	}
+
+	/**
+	 * Build item resource from a changeset
+	 *
+	 * @param ChangeSet $changeSet
+	 * @return array
+	 */
+	protected function getChangeSetResource(ChangeSet $changeSet) {
+		$hal = [
+			'_links' => [
+				'self' => [
+					'href' => $this->SetLink($changeSet->ID)
+				]
+			],
+			'ID' => $changeSet->ID,
+			'Name' => $changeSet->Name,
+			'Created' => $changeSet->Created,
+			'LastEdited' => $changeSet->LastEdited,
+			'State' => $changeSet->State,
+			'_embedded' => ['ChangeSetItems' => []]
+		];
+		foreach($changeSet->Changes() as $changeSetItem) {
+			/** @var ChangesetItem $changeSetItem */
+			$resource = $this->getChangeSetItemResource($changeSetItem);
+			$hal['_embedded']['ChangeSetItems'][] = $resource;
+		}
+		return $hal;
+	}
+
+	/**
+	 * Build item resource from a changesetitem
+	 *
+	 * @param ChangeSetItem $changeSetItem
+	 * @return array
+	 */
+	protected function getChangeSetItemResource(ChangeSetItem $changeSetItem) {
+		$hal = [
+			'_links' => [
+				'self' => [
+					'href' => $this->ItemLink($changeSetItem->ID)
+				]
+			],
+			'ID' => $changeSetItem->ID,
+			'Created' => $changeSetItem->Created,
+			'LastEdited' => $changeSetItem->LastEdited,
+			'Title' => $changeSetItem->getTitle(),
```

Comment:
Damian fixed that in a later commit already: https://github.com/open-sausages/silverstripe-framework/commit/628cc2f3834f47fce926360530d15696d3058bb3